src/components/auth-provider.tsx

```diff
@@ -199,20 +192,26 @@ function createWpcomClient(
 		return params;
 	};
 
-	return Object.assign( wpcom, {
-		request: function ( params: WpcomParams, callback: unknown ) {
-			const wrappedCallback = ( err: unknown, response: unknown, headers: unknown ) => {
-				if ( err ) {
-					void handleInvalidTokenError( err );
-				}
-				if ( typeof callback === 'function' ) {
-					callback( err, response, headers );
-				}
-			};
+	// Wrap the request handler to add locale and error handling before passing to wpcomFactory
+	const wrappedRequestHandler = (
+		params: object,
+		callback: ( err: unknown, response?: unknown, headers?: unknown ) => void
+	) => {
+		const modifiedParams = addLocaleToParams( params as WpcomParams );
+		const wrappedCallback = ( err: unknown, response: unknown, headers: unknown ) => {
+			if ( err ) {
+				void handleInvalidTokenError( err );
+			}
+			if ( typeof callback === 'function' ) {
+				callback( err, response, headers );
+			}
+		};
 
-			return originalRequestHandler( addLocaleToParams( params ), wrappedCallback );
-		},
-	} );
+		return wpcomXhrRequest( modifiedParams, wrappedCallback );
+	};
+
+	const wpcom = wpcomFactory( token, wrappedRequestHandler as typeof wpcomXhrRequest );
```

Comment:
```suggestion
	const wpcom = wpcomFactory( token, wrappedRequestHandler );
```
Doesn't look like we need the `as` operator here.

Reply:
Thanks, removed.